### PR TITLE
[minio-cpp] Update to 0.2.0

### PR DESCRIPTION
**Pull request by kobalicek**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/minio-cpp/portfile.cmake [hunk 1]
```diff
@@ -1,21 +1,26 @@
...
-
-vcpkg_cmake_install()
-vcpkg_copy_pdbs()
-vcpkg_fixup_pkgconfig()
-file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
-file(INSTALL "${SOURCE_PATH}/LICENSE" DESTINATION "${CURRENT_PACKAGES_DIR}/share/${PORT}" RENAME copyright)
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO minio/minio-cpp
+    REF v0.2.0
```
Comment (Osyotr) [inline ports/minio-cpp/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"
```

File: ports/minio-cpp/portfile.cmake [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
+vcpkg_cmake_config_fixup(
+    PACKAGE_NAME miniocpp
+    CONFIG_PATH "lib/cmake/miniocpp")
+vcpkg_copy_pdbs()
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE
+    "${CURRENT_PACKAGES_DIR}/debug/include")
+
+file(INSTALL "${SOURCE_PATH}/LICENSE"
```
Comment (Osyotr) [inline ports/minio-cpp/portfile.cmake:24]: `vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE")`

File: ports/minio-cpp/vcpkg.json [hunk 3]
```diff
@@ -24,6 +23,10 @@
     {
       "name": "vcpkg-cmake-config",
       "host": true
+    },
+    {
+      "name": "zlib",
+      "platform": "windows"
```
Comment (Osyotr) [inline ports/minio-cpp/vcpkg.json:29]: Upstream does unconditional `find_package(ZLIB REQUIRED)`, so platform expression should be removed.

**Comment by kobalicek** [comment]
CI seems stuck

**Review by Osyotr — CHANGES_REQUESTED** [review]

File: ports/minio-cpp/vcpkg.json (hunk 3 above)
Comment (dg0yt) [reply ports/minio-cpp/vcpkg.json:29]: Always use the vcpkg version.

File: ports/minio-cpp/portfile.cmake [hunk 4]
```diff
@@ -1,21 +1,24 @@
...
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(
+    PACKAGE_NAME miniocpp
+    CONFIG_PATH "lib/cmake/miniocpp")
+vcpkg_copy_pdbs()
+vcpkg_fixup_pkgconfig()
+
+file(REMOVE_RECURSE
```
Comment (Osyotr) [inline ports/minio-cpp/portfile.cmake:21]: Nit: don't break this line so early

File: ports/minio-cpp/portfile.cmake [hunk 5]
```diff
@@ -1,21 +1,24 @@
...
+
+vcpkg_cmake_configure(
+    SOURCE_PATH "${SOURCE_PATH}"
+    DISABLE_PARALLEL_CONFIGURE
+)
+
+vcpkg_cmake_install()
+vcpkg_cmake_config_fixup(
+    PACKAGE_NAME miniocpp
+    CONFIG_PATH "lib/cmake/miniocpp")
```
Comment (Osyotr) [inline ports/minio-cpp/portfile.cmake:17]: Nit: put `)` at new line or make whole statement one line

**Review by Osyotr — APPROVED** [review]

**Review by BillyONeal — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for the update!

**Pull request merged**